### PR TITLE
add locale property to resource entries in lock file

### PR DESCRIPTION
**Pull request by analogrelay**
I wasn't able to find an easy place to add a test... basically, you can see this issue when trying to restore `Microsoft.Data.OData` version `5.6.4`.

Lock File before this change:

``` json
      "Microsoft.Data.OData/5.6.4": {
        "type": "package",
        "dependencies": {
          "Microsoft.Data.Edm": "[5.6.4]",
          "System.Spatial": "[5.6.4]"
        },
        "compile": {
          "lib/portable-net45+wp8+win8+wpa/Microsoft.Data.OData.dll": {}
        },
        "runtime": {
          "lib/portable-net45+wp8+win8+wpa/Microsoft.Data.OData.dll": {}
        },
        "resource": {
          "lib/portable-net45+wp8+win8+wpa/de/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/es/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/fr/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/it/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/ja/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/ko/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/ru/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/zh-Hans/Microsoft.Data.OData.resources.dll": {},
          "lib/portable-net45+wp8+win8+wpa/zh-Hant/Microsoft.Data.OData.resources.dll": {}
        }
      },
```

Lock File after this change:

``` json
      "Microsoft.Data.OData/5.6.4": {
        "type": "package",
        "dependencies": {
          "Microsoft.Data.Edm": "[5.6.4]",
          "System.Spatial": "[5.6.4]"
        },
        "compile": {
          "lib/portable-net45+wp8+win8+wpa/Microsoft.Data.OData.dll": {}
        },
        "runtime": {
          "lib/portable-net45+wp8+win8+wpa/Microsoft.Data.OData.dll": {}
        },
        "resource": {
          "lib/portable-net45+wp8+win8+wpa/de/Microsoft.Data.OData.resources.dll": {
            "locale": "de"
          },
          "lib/portable-net45+wp8+win8+wpa/es/Microsoft.Data.OData.resources.dll": {
            "locale": "es"
          },
          "lib/portable-net45+wp8+win8+wpa/fr/Microsoft.Data.OData.resources.dll": {
            "locale": "fr"
          },
          "lib/portable-net45+wp8+win8+wpa/it/Microsoft.Data.OData.resources.dll": {
            "locale": "it"
          },
          "lib/portable-net45+wp8+win8+wpa/ja/Microsoft.Data.OData.resources.dll": {
            "locale": "ja"
          },
          "lib/portable-net45+wp8+win8+wpa/ko/Microsoft.Data.OData.resources.dll": {
            "locale": "ko"
          },
          "lib/portable-net45+wp8+win8+wpa/ru/Microsoft.Data.OData.resources.dll": {
            "locale": "ru"
          },
          "lib/portable-net45+wp8+win8+wpa/zh-Hans/Microsoft.Data.OData.resources.dll": {
            "locale": "zh-Hans"
          },
          "lib/portable-net45+wp8+win8+wpa/zh-Hant/Microsoft.Data.OData.resources.dll": {
            "locale": "zh-Hant"
          }
        }
      },
```

/cc @davidfowl @pakrym @emgarten @joelverhagen @yishaigalatzer

### CODE REVIEW

**Comment by davidfowl** [comment]
Needs a test 
/cc @emgarten

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs [hunk 1]
```diff
@@ -317,7 +317,13 @@ internal static class LockFileUtils
                 {
                     foreach (var item in group.Items)
                     {
-                        yield return new LockFileItem(item.Path);
+                        var newItem = new LockFileItem(item.Path);
+                        object locale;
+                        if (item.Properties.TryGetValue("locale", out locale))
+                        {
+                            newItem.Properties["locale"] = locale as string;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs]: (string)locale  since this should always be a string

**Comment by emgarten** [comment]
:+1: on tests, make this round trips as expected and that LockFile.Equals handles it appropriately

**Comment by analogrelay** [comment]
:up: :date:

Tests!

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**